File: core/src/main/java/com/ustadmobile/core/view/ContentEntryListView.java
package com.ustadmobile.core.view;

import com.ustadmobile.core.db.UmProvider;
import com.ustadmobile.lib.db.entities.ContentEntryWithStatusAndMostRecentContainerUid;
import com.ustadmobile.lib.db.entities.DistinctCategorySchema;
import com.ustadmobile.lib.db.entities.Language;

import java.util.List;
import java.util.Map;

public interface ContentEntryListView extends UstadView {

    String VIEW_NAME = "ContentEntryList";

    void setContentEntryProvider(UmProvider<ContentEntryWithStatusAndMostRecentContainerUid> entryProvider);

    void setToolbarTitle(String title);

    void showError();

    void setCategorySchemaSpinner(Map<Long, List<DistinctCategorySchema>> spinnerData);

    void setLanguageOptions(List<Language> result);
}

File: core/src/main/java/com/ustadmobile/core/view/ContentEntryListView.kt
package com.ustadmobile.core.view

import com.ustadmobile.core.db.UmProvider
import com.ustadmobile.lib.db.entities.ContentEntryWithStatusAndMostRecentContainerUid
import com.ustadmobile.lib.db.entities.DistinctCategorySchema
import com.ustadmobile.lib.db.entities.Language

interface ContentEntryListView : UstadView {

    fun setContentEntryProvider(entryProvider: UmProvider<ContentEntryWithStatusAndMostRecentContainerUid>)

    fun setToolbarTitle(title: String)

    fun showError()

    fun setCategorySchemaSpinner(spinnerData: Map<Long, List<DistinctCategorySchema>>)

    fun setLanguageOptions(result: List<Language>)

    companion object {

        const val VIEW_NAME = "ContentEntryList"
    }
}
